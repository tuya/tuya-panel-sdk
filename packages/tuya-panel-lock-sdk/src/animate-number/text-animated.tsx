import React, { Component } from 'react';
import { Animated } from 'react-native';
import { TYText } from 'tuya-panel-kit';
import { TextInputIProp } from './interface';

export class TextInputForAnimated extends Component<TextInputIProp> {
  render() {
    const { index, displayValues, ...others } = this.props;
    return <TYText {...others}>{displayValues[Math.floor(index)]}</TYText>;
  }
}

const AnimatedText = Animated.createAnimatedComponent(TextInputForAnimated);

export default AnimatedText;
