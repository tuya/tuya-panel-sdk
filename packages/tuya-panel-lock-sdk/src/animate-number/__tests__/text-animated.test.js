import React from 'react';
import { shallow } from 'enzyme';
import { TextInputForAnimated } from '../text-animated';

describe('TextInputForAnimated components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <TextInputForAnimated
        style={{ color: '#F00' }}
        index={1}
        displayValues={[8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
