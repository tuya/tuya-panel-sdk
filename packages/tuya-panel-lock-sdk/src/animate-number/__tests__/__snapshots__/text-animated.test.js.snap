// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputForAnimated components basic render 1`] = `
<TYText
  align={null}
  color={null}
  size={null}
  style={
    Object {
      "color": "#F00",
    }
  }
  type={null}
  weight={null}
>
  9
</TYText>
`;
